write a very simple getting started guide (#427)
kubernetes-sigs/service-apis: docs-src/getting-started.md

# Getting started with Services APIs

## Installing CRDs

Service APIs is a collection of Kubernetes Custom Resource Definitions that need

> to be installed into any Kubernetes cluster.

Reviewer: Maybe "This project provides a collection of Custom Resource Definitions that can be installed into any Kubernetes 1.16+ cluster."

To install the CRDs, please execute:

```
kubectl kustomize "github.com/kubernetes-sigs/service-apis/config/crd?ref=master" \

> | kubectl apply -f -

Reviewer: Is there a good way to do this without kustomize? It seems like we're not really using kustomize for anything. (This does not need to block this PR).

Author: We do this in our Makefile too: https://github.com/kubernetes-sigs/service-apis/blob/master/Makefile#L54

Do note that the user doesn't need to install `kustomize`. It is included as a plugin inside `kubectl` itself. Users need to make sure that their `kubectl` version is not an ancient one, which again is expected.

```

## Install an implementation

> Service APIs is a set of specification that multiple implementations confirm to.

Reviewer: This doesn't feel quite right. Maybe "Multiple projects implement the APIs defined by this project."

You will need to either install an implementation or verify if one is already
setup for your cluster.

## Sample Gateway

Once you have the CRDs and an implementation installed, you are ready to
use Service APIs.

In this example, we are installing three resources:

> - An `acme-lb` GatewayClass which is being managed by a `acme.io/gateway-controller` controller running in the cluster.

Reviewer: Nit: This line feels too long.

- A Gateway which is of type `acme-lb`:
    - This gateway has a single HTTP listener on port 80 which selects HTTPRoutes
      from all namespaces which have the label `app: foo` on them.

- Finally, we have an HTTPRoute resource which is attached to the above Gateway
  and has two rules:

>     - All requests with Path beginning with `/bar` are forwarded to my-service1

Reviewer: Nit: I'm not sure path needs to be capitalized here. I think it could use an `a` before it though.

      Kubernetes Service.

>     - All requests with Path beginning with `/some/thing` AND have an HTTP header

Reviewer: Same comment as above

# Getting started with Services APIs

## Installing CRDs

> This project provides a collection of Custom Resource Definitions (CRDs)that can

Reviewer: Nit:
```suggestion
This project provides a collection of Custom Resource Definitions (CRDs) that can
```

be installed into any Kubernetes (>= 1.16) cluster.

To install the CRDs, please execute:

```
kubectl kustomize "github.com/kubernetes-sigs/service-apis/config/crd?ref=master" \
| kubectl apply -f -
```

## Install an implementation

Multiple projects implement the APIs defined by this project.

> You will need to either install an implementation or verify if one is already

Reviewer: Nit
```suggestion
You will need to either install an implementation or verify that one is already
```

setup for your cluster.

## Sample Gateway

Once you have the CRDs and an implementation installed, you are ready to
use Service APIs.

In this example, we are installing three resources:

- An `acme-lb` GatewayClass which is being managed by a `acme.io/gateway-controller`

>   controller running in the cluster.

Reviewer: Maybe should add a note that users may want to use a GatewayClass provided by the implementation installed in their cluster. If one is not provided, they should ensure that the controller name used here matches a controller name supported by an implementation installed in their cluster.

Author: > If one is not provided

Skipped this part. We shouldn't be encouraging implementers not providing a GatewayClass right now.

  the implementation and must be installed in the cluster.
- A Gateway which is of type `acme-lb`:
    - This gateway has a single HTTP listener on port 80 which selects HTTPRoutes
      from all namespaces which have the label `app: foo` on them.

- Finally, we have an HTTPRoute resource which is attached to the above Gateway
  and has two rules:
    - All requests with path beginning with `/bar` are forwarded to my-service1
      Kubernetes Service.
    - All requests with path beginning with `/some/thing` AND have an HTTP header
      `magic: foo` are forwarded to my-service2 Kubernetes Service.

> With this configuration, you now have a Gateway reource which is forwarding

Reviewer: ```suggestion
With this configuration, you now have a Gateway resource which is forwarding
```

```

> {% include 'basic-http.yaml' %}

Reviewer: I don't have a good solution for this, and this doesn't need to block the PR, but one downside of embedding examples is that the examples could actually begin to skew from the docs. Maybe we should find a way to document which examples are being used in which docs?

Author: I thought about this problem too and arrived at the same solution as you.
Essentially adding a link to the guide file as a yaml comment.
Let's continue this in https://github.com/kubernetes-sigs/service-apis/issues/446.

```

> For more advanced examples, please checkout other [guides](guides.md).

Reviewer: Tiny nit:
```suggestion
For more advanced examples, please read the other [guides](guides.md).
```

For more advanced examples, please read the other [guides](guides.md).

## Uninstalling the CRDs

> 

Reviewer: Maybe add a bit of an explanation here, something like:

```
To uninstall the CRDs and all resources created with them, run the following command. Note that this will remove all GatewayClass and Gateway resources in your cluster. If you have been using these resources for any other purpose, do not uninstall these CRDs.
```
